### DISCUSSION TITLE
Best Practice To Making appwrite function as fast as possible.

### DISCUSSION THREAD

**Discussion by issasaleh** [appwrite/appwrite#5882] [Q&A]
Hi,

What steps do you think I should follow to make the function execute Very fast?

thanks,

**Comment by stnguyen90** [comment]
Several factors can affect how long an execution takes including:

1. startup
2. network latency
3. inefficiencies in code

For startup, deployment size can affect the transfer of the code. In addition, the language can affect startup times. For example, JVMs can take longer to startup, while compiled code can be faster to execute.

If you're making network calls in your function code, the network latency of those network calls can slow down function executions.

Lastly, inefficiencies in code, such as O(n^2) operations or executing code serially when it can be done concurrently, can make execution take longer.

These suggestions apply to software in general, and you can probably find more things to analyze by searching online. Before you dive too deep into this, though, you should measure the performance of your function executions to see what parts are slow and what's worth optimizing.
